### ISSUE TITLE
Envoy GRPC v2 api. 

### ISSUE THREAD

**Issue by skwasiborski** [Microsoft/service-fabric-reverse-proxy#7]
I made an implementation of v2 envoy control plane api [https://github.com/skwasiborski/csharp-control-plane](https://github.com/skwasiborski/csharp-control-plane). Would you be interested in using the api in this project?

**Comment by mkosieradzki** [comment]
@skwasiborski It's already implemented (correct me if I am wrong) in this branch (I have initially missed it as well): https://github.com/Microsoft/service-fabric-reverse-proxy/tree/kavyako-v2api

**Comment by skwasiborski** [comment]
Thanks @mkosieradzki - missed that. It is initial implementation. It seems to be not working currently e.g. it exposes only ADS service but configures XDS for cluster and route configuration.

I am also considering making a Revers Proxy for mesh model where all communication is done Proxy<->Proxy or within the same machine (like in Istio or recommended Envoy deployments). I will see if it would be possible to extend this implementation with such capability.

**Comment by mkosieradzki** [comment]
@skwasiborski awesome, good luck! Maybe you should contribute to this repo?

Meanwhile I have started implementation of gRPC middleware for ASP.NET Core (to avoid using native-code google gRPC server). However I don't believe the full gRPC support will be ready before .NET Core 2.2 (as the current HTTP/2 support in Kestrel is not ready). However gRPC-Web support should be ready much earlier.

**Comment by skwasiborski** [comment]
I managed to put up a very initial working version of setup where all communication is done from Envoy to Envoy. You can play with it here https://github.com/skwasiborski/ServiceFabricEnvoyReverseProxy.

If it is of interest to this project I can try to rewrite it in a way that matches this project.

**Comment by mkosieradzki** [comment]
@skwasiborski Congratulations. Great job!

**Comment by skwasiborski** [comment]
Thanks. It is super early version with lots of hacks and things to correct. It also currently is not easily deployable to any cluster. The control plane and data plane are to distinct services which I am not sure is most convenient setup (this project packages everything in single container).

BTW. I am getting some success with running envoy on windows in hyperv isolation mode. I managed to get it working outside of SF and now I am trying to force SF to use successfully use patched docker daemon.

**Comment by kavyako** [comment]
@skwasiborski just catching up on this thread. awesome! 
I had started the c# control plane with envoy v2 apis in this branch https://github.com/Microsoft/service-fabric-reverse-proxy/tree/kavyako-v2api, this is a port of the java control plane but got pulled into something else. This is a very initial prototype and yet to be tested.
Looks like you have a port of the go control plane in C#. 
We are currently using the V1 api and will start focusing more on the v2 apis end of august. We could definitely look at combining the efforts.

**Comment by skwasiborski** [comment]
@kavyako yes I based my implementation on the go version. It seemed to be more used and more mature. I managed to pull all the pieces together in the second repo I linked. If you are interested in some joint work drop me an email I can adjust my port to your needs or integrated into the codebase.

**Comment by kavyako** [comment]
+ @vturecek
